### PR TITLE
WIP: Allow different json media types for params, requests, and responses

### PR DESCRIPTION
**Pull request by ckeboss**
Addresses https://github.com/cdimascio/express-openapi-validator/issues/143

### CODE REVIEW

File: src/middlewares/ajv/index.ts [hunk 1]
```diff
@@ -95,17 +93,5 @@ function createAjv(
     );
   }
 
-  if (openApiSpec.components.requestBodies) {
```
Comment (cdimascio) [inline src/middlewares/ajv/index.ts]: We shouldn't remove this addSchema section as we need to add shemas any `components.requestBodies` 
for references: https://swagger.io/docs/specification/components/

Comment (ckeboss) [reply src/middlewares/ajv/index.ts]: I believe this is handled by AJV when we pass the request body.

Also, all the test work without it? Is there a specific test I could add to verify this?

**Comment by ckeboss** [comment]
We could use both https://www.npmjs.com/package/media-typer and https://www.npmjs.com/package/content-type together, as they do the same thing, but split into multiple libraries.

**Comment by cdimascio** [comment]
Cool. Let's go with  https://www.npmjs.com/package/media-typer and https://www.npmjs.com/package/content-type

File: src/middlewares/ajv/index.ts (hunk 1 above)
Comment (cdimascio) [reply src/middlewares/ajv/index.ts]: Good point. You're right. This is tested in request.bodies.ref.spec.ts. Looks like its safe to remove

**Comment by cdimascio** [comment]
@ckeboss i've merged in the v3.0.0 codebase into your PR branch.

It will be fantastic if you can make the changes to use [media-types](https://www.npmjs.com/package/media-typer) and [content-type](https://www.npmjs.com/package/content-type) instead of media-type.

Looking forward to getting this in!
Thanks!

**Comment by ckeboss** [comment]
Thanks. Yes, I'll make those changes today

**Comment by cdimascio** [comment]
@ckeboss i've merged master into your PR

**Comment by ckeboss** [comment]
Thank you, sorry, been busy, will change out those libraries soon.

**Comment by ckeboss** [comment]
@cdimascio updated the media type / content type parser libraries.

**Comment by cdimascio** [comment]
Thanks a lot @ckeboss. Will have a look soon!

**Comment by cdimascio** [comment]
@all-contributors add @ckeboss for code and test

**Pull request merged**